mlflow/pipelines/steps/split.py

```diff
@@ -191,7 +202,22 @@ def _run(self, output_directory):
                 importlib.import_module(post_split_module_name), post_split_fn_name
             )
             _logger.debug(f"Running {post_split_fn_name} on train, validation and test datasets.")
-            (train_df, validation_df, test_df) = post_split(train_df, validation_df, test_df)
+            (post_split_train_df, post_split_validation_df, post_split_test_df) = post_split(
+                train_df, validation_df, test_df
+            )
+            if (
+                is_subset(post_split_train_df, train_df)
+                and is_subset(post_split_validation_df, validation_df)
+                and is_subset(post_split_test_df, test_df)
+            ):
+                train_df = post_split_train_df
+                validation_df = post_split_validation_df
+                test_df = post_split_test_df
+            else:
+                raise MlflowException(
+                    f"Attempting to perform arbitrary ETL transform in {post_split_fn_name}. "
+                    f"Arbitrary transformations should go into the transform step."
+                )
```

Comment:
Btw this could break existing pipeline jobs for some customers. So we might need to note this as part of breaking change maybe. 
cc @dbczumar to confirm if this is a breaking change and if yes, how do we proceed with merging this?

Reply:
MLP is experimental - we can break things if we have good motivation for doing so